docs/ancestry.md

```diff
@@ -961,11 +961,150 @@ section for more details.
 
 ### Migration events
 
-```{eval-rst}
-.. todo:: examples of using the record_migrations
 
+```{code-cell}
+:tags: [remove-cell]
+
+    msprime.core.set_seed_rng_seed(42)
 ```
 
+When running simulations with {ref}`demography <sec_demography>`, by
+default, migration events are not recorded in the resulting tree sequence.
+While the population associated with each node is determined by the simulated migration
+events, knowledge of the
+precise sequence of migrations is useful in a number of situations: for example see
+the `tskit`
+[tutorial on introgression.](<https://tskit-dev.github.io/tutorials/introgression.html>)
+
+The `record_migrations` parameter allows us to record these events in the
+simulated tree sequence outputted by {func}`.sim_ancestry`. Briefly, a migration record
+encodes the node associated with the migration, the time of the event, the source and
+destination populations (backwards in time), and the left and right coordinates of the
+migrating segement. For more details on migration records, see the
+[migration table definition](<https://tskit.dev/tskit/docs/stable/data-model.html#sec-migration-table-definition>).
+
+Here, we provide a simple example of the effect of setting `record_migrations=True`
+using the {meth}`stepping stone model<msprime.Demography.stepping_stone_model>`
+where migration is permitted between adjacent populations. Additionally, we
+set `record_full_arg=True` (see {ref}`previous section <sec_ancestry_full_arg>`)
+to record nodes corresponding to migration events. This is not necessary, but will be
+helpful to visualise the result.
+
+```{code-cell}
+	N = 10
+	demography = msprime.Demography.stepping_stone_model(
+		[20] * N, migration_rate=0.1, boundaries=True
+	)
+	ts = msprime.sim_ancestry(
+		{0: 1, N - 1: 1},
+		demography=demography,
+		record_migrations=True,
+		record_full_arg=True,
+		random_seed=6,
+	)
+```
+
+In the following visualisation, we trace the migration history of two samples backwards
+in time to their most recent common ancestor. Each one of the ten populations in the
+model is assigned a 
+colour and a position on the x axis. The time of migration events is recorded on the y
+axis.
+
+```{code-cell}
+    :"tags": ["hide-input"]
+    import matplotlib
+    import numpy as np
+
+    cmap = matplotlib.cm.get_cmap('viridis')
+    tree = ts.first()
+    mig_color_dict = {}
+    for i, linestyle in zip(ts.samples()[::2], ["dotted", "dashdot"]):
+        mig = ts.tables.migrations
+        loc = []
+        time = []
+        while i != -1:
```

Comment:
-1 -> tskit.NULL

Reply:
It's not helpful to use ``i`` as a node id here and also as a generic index variable just below. We usually use ``u`` or ``v`` as node variables.